algosdk/abi/util.py

```diff
@@ -1,23 +1,24 @@
 import re
 
-from .uint_type import UintType
-from .ufixed_type import UfixedType
-from .byte_type import ByteType
-from .bool_type import BoolType
-from .address_type import AddressType
-from .string_type import StringType
-from .array_dynamic_type import ArrayDynamicType
-from .array_static_type import ArrayStaticType
-from .tuple_type import TupleType
-from .. import error
+from algosdk.abi.base_type import ABIType
+from algosdk.abi.uint_type import UintType
+from algosdk.abi.ufixed_type import UfixedType
+from algosdk.abi.byte_type import ByteType
+from algosdk.abi.bool_type import BoolType
+from algosdk.abi.address_type import AddressType
+from algosdk.abi.string_type import StringType
+from algosdk.abi.array_dynamic_type import ArrayDynamicType
+from algosdk.abi.array_static_type import ArrayStaticType
+from algosdk.abi.tuple_type import TupleType
+from algosdk import error
 
 
 # Globals
 UFIXED_REGEX = r"^ufixed([1-9][\d]*)x([1-9][\d]*)$"
 STATIC_ARRAY_REGEX = r"^([a-z\d\[\](),]+)\[([1-9][\d]*)]$"
 
 
-def type_from_string(s):
+def type_from_string(s: str) -> ABIType:
```

Comment:
Not directly related to this PR, but in the sprit of improving ABI support, I realized it is possible to implement this as a static method on the `ABIType` class, like `ABIType.of()` or `ABIType.from_string()` (not sure what the best name would be for Python).

You can do it putting all the subclass imports that would otherwise be recursive inside the method body. Something like:

```python
@staticmethod
def of(s: str) -> "ABIType":
   from algosdk.abi.uint_type import UintType
   from algosdk.abi.uint_type import UintType
   ...
```

This avoids the circular import issue.

Reply:
Amazing 🙌  I like it

Reply:
+1 on `from_string(F)`